### PR TITLE
Ship vulture as a package.

### PR DESCRIPTION
**Pull request by RJ722**
Provide vulture as a package rather than as a script
## Description

New directory structure:
```
├── LICENSE.txt
├── MANIFEST.in
├── NEWS.rst
├── README.rst
├── TODO.rst
├── build
│   ├── bdist.macosx-10.12-x86_64
│   └── lib
│       └── vulture
│           ├── __init__.py
│           └── core.py
├── dist
│   ├── vulture-0.14
│   │   ├── LICENSE.txt
│   │   ├── MANIFEST.in
│   │   ├── NEWS.rst
│   │   ├── PKG-INFO
│   │   ├── README.rst
│   │   ├── TODO.rst
│   │   ├── setup.cfg
│   │   ├── setup.py
│   │   ├── tests
│   │   │   ├── __init__.py
│   │   │   ├── test_errors.py
│   │   │   ├── test_format_strings.py
│   │   │   ├── test_imports.py
│   │   │   ├── test_scavenging.py
│   │   │   └── test_script.py
│   │   ├── tox.ini
│   │   ├── vulture
│   │   │   ├── __init__.py
│   │   │   ├── core.py
│   │   │   └── whitelists
│   │   │       ├── __init__.py
│   │   │       └── stdlib.py
│   │   └── vulture.egg-info
│   │       ├── PKG-INFO
│   │       ├── SOURCES.txt
│   │       ├── dependency_links.txt
│   │       ├── entry_points.txt
│   │       └── top_level.txt
│   ├── vulture-0.14-py2.7.egg
│   ├── vulture-0.14.tar
│   ├── vulture-0.14.tar.gz
│   └── vulture-0.14.zip
├── htmlcov
│   ├── coverage_html.js
│   ├── index.html
│   ├── jquery.ba-throttle-debounce.min.js
│   ├── jquery.hotkeys.js
│   ├── jquery.isonscreen.js
│   ├── jquery.min.js
│   ├── jquery.tablesorter.min.js
│   ├── keybd_closed.png
│   ├── keybd_open.png
│   ├── status.json
│   ├── style.css
│   ├── vulture___init___py.html
│   ├── vulture_core_py.html
│   ├── vulture_py.html
│   ├── vulture_whitelists___init___py.html
│   └── vulture_whitelists_stdlib_py.html
├── release.sh
├── setup.cfg
├── setup.py
├── tests
│   ├── __init__.py
│   ├── test_errors.py
│   ├── test_format_strings.py
│   ├── test_imports.py
│   ├── test_scavenging.py
│   └── test_script.py
├── tox.ini
├── vulture
│   ├── __init__.py
│   ├── core.py
│   └── whitelists
│       └── stdlib.py
└── vulture.egg-info
    ├── PKG-INFO
    ├── SOURCES.txt
    ├── dependency_links.txt
    ├── entry_points.txt
    └── top_level.txt
```

## Related Issue

Closes: #52 

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation in the README file.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -45,8 +45,11 @@ def run_tests(self):
...
     ],
     entry_points={
-        'console_scripts': ['vulture = vulture:main'],
+        'console_scripts': ['vulture = vulture.core:main'],
     },
     tests_require=['pytest', 'pytest-cov'],
     cmdclass={'test': PyTest},
+    packages=['vulture'],
+    package_dir={'vulture': 'vulture'},
+    package_data={'vulture': ['whitelists/**/*.py', ]},
```
Comment (RJ722) [inline setup.py]: Do these changes require a different commit under the whitelist PR?

Comment (RJ722) [inline setup.py]: Do these changes require a different commit under the whitelist PR?

File: tests/test_script.py [hunk 2]
```diff
@@ -6,30 +6,31 @@
...
 
 
 def test_script_without_whitelist():
-    assert call_vulture(['vulture.py']) == 1
+    assert call_vulture(['vulture/core.py']) == 1
 
 
 def test_exclude():
-    assert call_vulture(['vulture.py', '--exclude', 'vulture.py']) == 0
+    assert call_vulture(['vulture/core.py', '--exclude', 'vulture/core.py']) \
```
Comment (jendrikseipp) [inline tests/test_script.py]: you can use `'--exclude', 'core.py'` to make this fit onto one line

File: tests/test_script.py [hunk 3]
```diff
@@ -6,30 +6,31 @@
 
 DIR = os.path.dirname(os.path.abspath(__file__))
 REPO = os.path.dirname(DIR)
-WHITELIST = os.path.join(REPO, 'whitelists', 'stdlib.py')
+WHITELIST = os.path.join(REPO, 'vulture/whitelists', 'stdlib.py')
```
Comment (jendrikseipp) [inline tests/test_script.py]: `'vulture', 'whitelists'`

File: setup.py (hunk 1 above)
Comment (jendrikseipp) [inline setup.py]: Yes, let's not mix issues here.

File: MANIFEST.in [hunk 4]
```diff
@@ -2,4 +2,4 @@ include *.rst
 include *.txt
 include tests/*.py
 include tox.ini
-include whitelists/*.py
+include vulture/whitelists/*.py
```
Comment (jendrikseipp) [inline MANIFEST.in]: Commenting here since __init__.py is an empty file: I don't think we need `whitelists/__init__.py`, please remove it.

File: setup.py (hunk 1 above)
Comment (jendrikseipp) [inline setup.py]: drop comma: `['whitelists/**/*.py']`

File: vulture/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+__version__ = '0.14'
```
Comment (jendrikseipp) [inline vulture/__init__.py]: Please add `from core import Vulture` above __version__. As a side effect, we don't have to change the imports in the test files.

File: setup.py (hunk 1 above)
Comment (jendrikseipp) [inline setup.py]: To be clear: please add packages and package_dir in this PR and package_data in PR #50.

**Comment by RJ722** [comment]
Done making changes 😄 
I've updated the dir structure as well!

File: release.sh [hunk 6]
```diff
@@ -19,7 +19,7 @@ if [[ $retcode != 0 ]]; then
 fi
 
 # Bump version.
-sed -i -e "s/__version__ = '.*'/__version__ = '$VERSION'/" vulture.py
+sed -i -e "s/__version__ = '.*'/__version__ = '$VERSION'/" vulture/core.py
```
Comment (jendrikseipp) [inline release.sh]: The `__version__` attribute should be in `__init__.py` and only there.

File: vulture/__init__.py [hunk 7]
```diff
@@ -0,0 +1,2 @@
+from .core import Vulture   # noqa
```
Comment (jendrikseipp) [inline vulture/__init__.py]: I'd rather not use `noqa` tags in the code. Please use `assert Vulture` instead and leave an empty line after it to separate it from `__version__`.

**Comment by RJ722** [comment]
Should we add a test for checking the release mechanism also?

**Comment by jendrikseipp** [comment]
I don't think a test for this is needed.

File: vulture/__init__.py (hunk 5 above)
Comment (jendrikseipp) [inline vulture/__init__.py]: Please remove the version attribute from core.py

Comment (RJ722) [inline vulture/__init__.py]: I thought I had done it already 😕  Sorry!

Comment (RJ722) [inline vulture/__init__.py]: Done! 😄

**Comment by jendrikseipp** [comment]
I think the tests need to be updated.

**Comment by RJ722** [comment]
If our `__init__.py` is:
```
from .core import Vulture
assert Vulture

__version__ = '0.14'
```

Then this would lead to `__init.py__` importing from `core` and `core` importing `__version__` from `__init__.py`, Thus circular imports!

This can be prevented though, if our `__init__.py` file looks like:
```
__version__ = '0.14'

from .core import Vulture
assert Vulture
```

Is this correct?

**Comment by jendrikseipp** [comment]
Let's keep the version attribute in core.py and import it in `__init__.py`.

**Comment by RJ722** [comment]
Done!

File: vulture/core.py [hunk 8]
```diff
@@ -37,6 +37,7 @@
 
 __version__ = '0.14'
 
+
```
Comment (jendrikseipp) [inline vulture/core.py]: Delete empty line again?

File: vulture/__init__.py [hunk 9]
```diff
@@ -0,0 +1,3 @@
+from .core import __version__, Vulture
```
Comment (jendrikseipp) [inline vulture/__init__.py]: Use same order for import and asserts.

File: release.sh [hunk 10]
```diff
@@ -19,7 +19,7 @@ if [[ $retcode != 0 ]]; then
 fi
 
 # Bump version.
-sed -i -e "s/__version__ = '.*'/__version__ = '$VERSION'/" vulture.py
+sed -i -e "s/__version__ = '.*'/__version__ = '$VERSION'/" vulture/__init__.py
```
Comment (jendrikseipp) [inline release.sh]: Should change core.py

**Comment by RJ722** [comment]
Your level of observation amuses me! 😲

**Comment by jendrikseipp** [comment]
Thanks!

**Comment by RJ722** [comment]
My pleasure!
